lib/matplotlib/colorbar.py

```diff
@@ -239,28 +239,48 @@ def __init__(self, parent, userax=True):
                 parent._axes.add_child_axes(outer_ax)
                 outer_ax._axes.child_axes.remove(parent)
             else:
-                parent.remove()
+                try:
+                    parent.remove()
+                except ValueError:
+                    pass  # Already removed
         else:
             outer_ax = parent
 
+        # swap axes in the stack if its in there:
```

Comment:
I'm not sure how it wouldn't be there if you've created it above or gotten it passed in? But, wouldn't you want to `add(self)` regardless of whether the outer one is there or not?

Reply:
There was some test where we made a colorbar and it didn't actually ever get added. I can take ti out again to figure out which one...